### DISCUSSION TITLE
pymsql的import

### DISCUSSION THREAD

**Discussion by Qingelish** [nonebot/discussions#397] [问答]
为何我在nonebot插件种引用不了pymsql

**Comment by yanyongyu** [comment]
请提供详细信息

**Comment by Qingelish** [comment]
![mmexport1688705517241](https://github.com/nonebot/discussions/assets/19321118/6dca90cc-e9da-4ad4-9281-2237f47dc121)

**Comment by synodriver** [comment]
似乎是没有安装……注意大小写，然后就是在异步框架中使用同步库并不明智，推荐aiomysql或者asyncmy

**Reply by Qingelish** [reply]
突放下面了

**Comment by Qingelish** [comment]
![mmexport1688709010054](https://github.com/nonebot/discussions/assets/19321118/d8fc68d5-b8bc-4dc9-8ece-24a5a32f16ba)

直接运行，装是装了的。

我也理解了nonebot暂时没有链接外部库的方案

所以报错是非常正常的情况。

**Reply by yanyongyu** [reply]
你没有安装到虚拟环境内

**Reply by D-XH** [reply]
怎么装到虚拟环境内啊
